### PR TITLE
Clean up rust agent clippy warnings

### PR DESCRIPTION
**Pull request by jiangliu**
Clean up rust agent clippy warnings and simplify implementation.
The second patch touch the core logic, please help to do careful reviews:)

### CODE REVIEW

File: src/agent/src/grpc.rs [hunk 1]
```diff
@@ -103,20 +104,17 @@ impl agentService {
...
         // list) to bind mount all of them inside the container.
         let m = add_storages(sl!(), req.storages.to_vec(), &self.sandbox)?;
-        {
-            sandbox = self.sandbox.clone();
-            s = sandbox.lock().unwrap();
-            s.container_mounts.insert(cid.clone(), m);
-        }
+        // safe to unwrap() because we don't expect poisoned lock here.
+        let mut s = self.sandbox.lock().unwrap();
+        s.container_mounts.insert(req.container_id.clone(), m);
```
Comment (yyyeerbo) [inline src/agent/src/grpc.rs]: We did have some reason to limit the life of lock. rethink about it, otherwise there an be deablock

Comment (jiangliu) [reply src/agent/src/grpc.rs]: I will recheck the code:)
If needed , we could add a "drop(s)" to solve the possible deadlock.

Comment (jiangliu) [reply src/agent/src/grpc.rs]: I have checked the original codee again. 'sandbox' and 's'  are declared out side of the block scope, so the lock guard `s` won't be dropped when getting out of the block.
Anything I missed?

**Comment by tedyu** [comment]
lgtm

**Comment by egernst** [comment]
rebase needed.  Ping @jiangliu

**Comment by jiangliu** [comment]
We have more pending changes upon these changes, and we are still doing tests internally.
How about closing these PR and opening a new PR including all changes once it passes tests?

**Comment by grahamwhaley** [comment]
@jiangliu @yyyeerbo - what's the plan here - update this, or open a new PR with new set of commits?

**Comment by jiangliu** [comment]
> @jiangliu @yyyeerbo - what's the plan here - update this, or open a new PR with new set of commits?

I have had a talk with Li Fupan offline, and I will rebase the PR against the latest rust agent:)

**Comment by grahamwhaley** [comment]
:nudge: @jiangliu - time to rebase and update yet?

**Comment by jiangliu** [comment]
Just found the that we have moved onto 2.0-dev branch. So I will close this PR and create new PR against the 2.0-dev branch:)

**Pull request closed without merging**